docs/spec/lightclient/Lightclient_A_1.tla

```diff
@@ -0,0 +1,463 @@
+\* or the block is produced by a faulty primary.
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+FetchLightBlockInto(block, height) ==
+    IF IS_PRIMARY_CORRECT
+    THEN CopyLightBlockFromChain(block, height)
+    ELSE BC!IsLightBlockAllowedByDigitalSignatures(height, block)
+
+\* add a block into the light store    
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+LightStoreUpdateBlocks(lightBlocks, block) ==
+    LET ht == block.header.height IN    
+    [h \in DOMAIN lightBlocks \union {ht} |->
+        IF h = ht THEN block ELSE lightBlocks[h]]
+
+\* update the state of a light block      
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+LightStoreUpdateStates(statuses, ht, blockState) ==
+    [h \in DOMAIN statuses \union {ht} |->
+        IF h = ht THEN blockState ELSE statuses[h]]      
+
+\* Check, whether newHeight is a possible next height for the light client.
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+ScheduleTo(newHeight, pNextHeight, pTargetHeight, pLatestVerified) ==
+    LET ht == pLatestVerified.header.height IN
+    \/ /\ ht = pNextHeight
+       /\ ht < pTargetHeight
+       /\ pNextHeight < newHeight
+       /\ newHeight <= pTargetHeight
+    \/ /\ ht < pNextHeight
+       /\ ht < pTargetHeight
+       /\ ht < newHeight
+       /\ newHeight < pNextHeight
+    \/ /\ ht = pTargetHeight
+       /\ newHeight = pTargetHeight
+
+\* The loop of VerifyToTarget.
+\*
+\* TODO: add a traceability tag when Josef adds it in the English spec
+VerifyToTargetLoop ==
+      \* the loop condition is true
+    /\ latestVerified.header.height < TARGET_HEIGHT
+      \* pick a light block, which will be constrained later
+    /\ \E current \in BC!LightBlocks:
+        \* Get next LightBlock for verification
+        /\ IF nextHeight \in DOMAIN fetchedLightBlocks
+           THEN /\ LightStoreGetInto(current, nextHeight)
+                /\ UNCHANGED fetchedLightBlocks
+           ELSE /\ FetchLightBlockInto(current, nextHeight)
+                /\ fetchedLightBlocks' = LightStoreUpdateBlocks(fetchedLightBlocks, current)
+        \* Record that one more probe has been done (for complexity and model checking)
+        /\ nprobes' = nprobes + 1
+        \* Verify the current block
+        /\ LET verdict == ValidAndVerified(latestVerified, current) IN
+           \* Decide whether/how to continue
+           CASE verdict = "OK" ->
+              /\ lightBlockStatus' = LightStoreUpdateStates(lightBlockStatus, nextHeight, "StateVerified")
+              /\ latestVerified' = current
+              /\ state' =
+                    IF latestVerified'.header.height < TARGET_HEIGHT
+                    THEN "working"
+                    ELSE "finishedSuccess"
+              /\ \E newHeight \in HEIGHTS:
+                 /\ ScheduleTo(newHeight, nextHeight, TARGET_HEIGHT, current)
```

Comment:
This is nicer than english spec as we know where we are. So we could say: pick a height in (nextHeight, TARGET_HEIGHT]
```suggestion
                 /\ ScheduleTo(newHeight, nextHeight, TARGET_HEIGHT)
```

Reply:
Here you like a non-deterministic choice 👍